### DISCUSSION TITLE
Undefined array key 1

### DISCUSSION THREAD

**Discussion by sasori13L** [awcodes/filament-table-repeater#113] [Q&A] [answered]
Recently installed this plugin to my new Filament v3.2.0 project with Laravel 11. 

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TableRepeater::make('users')
                    ->headers([
                        Header::make('name')->width('150px'),
                    ])
                    ->schema([
                        Forms\Components\TextInput::make('asd'), // Plugin works if only one 
                        Forms\Components\TextInput::make('asdx'), // Undefined array key 1 error occurs if I add another component
                    ])
                    ->columnSpan('full')
            ]);
    }

**Comment by awcodes** [comment, marked as answer]
You have to have a header for each field in the schema.

**Comment by T3mmyKay** [comment]
Please I have this same issue

Forms\Components\Section::make('Products')->schema([
                    TableRepeater::make('purchaseOrderItems')
                        ->columnSpanFull()
                        ->stackAt(MaxWidth::Large)
                        ->headers([
                            Header::make('variation_id')->label('Product')
                                ->markAsRequired(),
                            Header::make('quantity')->label('Quantity'),
                            Header::make('unit_price')->label('Unit Price'),
                            Header::make('inline_total')->label('Inline Total'),
                        ])
                        ->live()
                        ->relationship('purchaseOrderItems')
                        ->hiddenLabel()
                        ->schema([
                            Select::make('variation_id')
                                ->relationship('variation')
                                ->label('Product')
                                ->searchable()
                                ->preload()
                                ->options(
                                    $variations->mapWithKeys(function (Variation $variation) {
                                        return [$variation->id => sprintf('%s (%s)', $variation->product->name, $variation->product->purchaseUnit->short_name)];
                                    })
                                )
                                ->afterStateUpdated(function (PurchaseOrder $record, Forms\Set $set, $state) {

                                    if ($state == null) {
                                        $set('unit_price', 0);
                                        $set('price', 0);
                                    }
                                    $variation_id = $state;
                                    $variation = Variation::find($variation_id);
                                    if ($variation) {

                                        $conversionFactor = $variation->product->purchaseUnit->base_unit_conversion ?? 1;
                                        $unitPrice = $variation->selling_price_inc_tax * $conversionFactor;
                                        $set('unit_price', $unitPrice);
                                        $set('price', $unitPrice);
                                    }
                                })
                                ->disableOptionsWhenSelectedInSiblingRepeaterItems()
                                ->required(),
                            Forms\Components\TextInput::make('quantity')
                                ->label('Quantity')
                                ->required()
                                ->live(true)
                                // After adding a new row, we need to update the totals
                                ->afterStateUpdated(function (Get $get, Set $set) {
                                    self::updateTotals($get, $set);
                                })
                                ->numeric()
                                ->default(1),
                            Forms\Components\TextInput::make('unit_price')
                                ->label('Unit Price')
                                ->required()
                                ->dehydrated(false)
                                ->prefix('₦')
                                ->numeric()
                                ->disabled(),
                            Forms\Components\Hidden::make('price'),
                            Forms\Components\Placeholder::make('inline_total')
                                ->hiddenLabel()
                                ->content(fn($get) => Number::currency(($get('quantity') * $get('unit_price')), 'NGN', 'en_NG'))
                        ])->addAction(
                            fn(Action $action) => $action->color('primary')->label('Add Product'),
                        )->deleteAction(
                            fn(Action $action) => $action->after(fn(Get $get, Set $set) => self::updateTotals($get, $set)),
                        ),
                        
                        The edit and view page keeps throwing out that error
                        
                        I don't know if it is because of  Forms\Components\Hidden::make('price'),

**Reply by awcodes** [reply]
Sorry, but this is too much to try to reason about. Can you submit a minimal reproduction repository?
